### PR TITLE
Bump numpy version in requirements_dev.

### PR DESCRIPTION
**Pull request by Carreau**
The comment in the file suggest that the numpy version should be the
latest one, it is not.

This would help to make sure that there is no regression in numpy,
and/or that the test suite would catch numpy deprecated functionality.

* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] AppVeyor and Travis CI passes
* [ ] Test coverage is 100% (Coveralls passes)

### CODE REVIEW

File: requirements_dev_numpy.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Break this out into a separate file to allow testing against
 # different versions of numpy. This file should pin to the latest
 # numpy version.
-numpy==1.17.2; python_version > '3.0'
+numpy==1.19.2; python_version > '3.0'
```
Comment (Carreau) [inline requirements_dev_numpy.txt:4]:
```suggestion
numpy==1.19.2; python_version > '3.6'
numpy==1.17.2; python_version <= '3.6'
```

**Comment by joshmoore** [comment]
@Carreau thoughts on the likes of https://dependabot.com to auto-bump?

**Comment by Carreau** [comment]
> @Carreau thoughts on the likes of [dependabot.com](https://dependabot.com) to auto-bump?

Why not, though I think it work great for apps (be sure your' always on the latest), but less for libraries, where you may want to be looser on your requirements. So it will depends on whether it tryed to bump everything everywhere or "just" in requirements.txt for CI.

**Comment by Carreau** [comment]
closing and reopening to kick tests.

**Pull request merged**